homeassistant/components/mqtt/__init__.py

```diff
@@ -370,18 +389,30 @@ def async_publish_service(call):
         msg_topic = call.data[ATTR_TOPIC]
         payload = call.data.get(ATTR_PAYLOAD)
         payload_template = call.data.get(ATTR_PAYLOAD_TEMPLATE)
+        payload_file_path = call.data.get(ATTR_PAYLOAD_FILE_PATH)
         qos = call.data[ATTR_QOS]
         retain = call.data[ATTR_RETAIN]
-        try:
-            if payload_template is not None:
+        if payload_template is not None:
+            try:
                 payload = \
                     template.Template(payload_template, hass).async_render()
-        except template.jinja2.TemplateError as exc:
-            _LOGGER.error(
-                "Unable to publish to '%s': rendering payload template of "
-                "'%s' failed because %s",
-                msg_topic, payload_template, exc)
-            return
+            except template.jinja2.TemplateError as exc:
+                _LOGGER.error(
+                    "Unable to publish to '%s': rendering payload template of "
+                    "'%s' failed because %s",
+                    msg_topic, payload_template, exc)
+                return
+        elif payload_file_path is not None:
```

Comment:
Thanks Pascal, that's clear to me. I was just pointing out that I was building on top of a situation I hadn't created myself: the service is already there and it still looks incomplete and ambiguous to me, but ok, be it the way it is.

Just a quick clarification: so from a component the call would be
`hass.data['mqtt'].async_publish(msg_topic, payload, qos, retain)`
?